src/engine/Execute.v3

```diff
@@ -32,10 +32,12 @@ component Execute {
 	}
 }
 
+type SuspendInfo(tag: Tag, vals: Array<Value>) #unboxed;
+
 type Result {
 	case Value(vals: Array<Value>) {}
 	case Throw(thrown: Throwable) {}
-	case Suspend(stack: WasmStack, exntag: Tag, valrefs: Range<Value>);
+	case Suspend(stack: WasmStack, info: SuspendInfo);
```

Comment:
I'm wondering if the combination of `(stack, info)` should just be a continuation reference. wdyt?

Reply:
I think this is a good replacement for `stack`, although I'm not sure how to handle the returned `tag` and `vals` by the suspend instruction. I'm thinking since after executing something like `resume $ct` where `$ct : [t1*] -> [t2*]`, `$ct` will need to suspend with a value array of `[t2*]` which is then pushed onto the parent stack. I'm wondering if this value array should be kept separate from the continuation reference as it is part of the information associated with that specific `suspend` rather than a part of the returned continuation itself.

Reply:
Oh, yes, I think you're right.